logger/test_log_file_merger.py

```diff
@@ -0,0 +1,153 @@
+"""
+Log file merger unit tests.
+"""
+
+import pathlib
+import shutil
+import pytest
+
+from .modules import log_file_merger
+
+
+TEMP_LOGS_DIRECTORY = pathlib.Path("logs_temp")
+MERGED_LOGS_FILENAME = pathlib.Path("merged_logs")
+UNSORTED_LOG_ENTRIES = [
+    "12:59:28: [INFO] [foo1.py | foo1 | 43] Foo1 initialized\n",
+    "13:00:04: [ERROR] [foo1.py | foo1 | 30] Foo1 could not be created\n",
+    "13:00:22: [ERROR] [foo1.py | foo1 | 49] Foo1 failed to create class object\n",
+    "12:59:40: [INFO] [foo2.py | foo2 | 43] Foo2 initialized\n",
+    "13:00:06: [ERROR] [foo2.py | foo2 | 30] Foo2 could not be created\n",
+    "13:00:09: [ERROR] [foo2.py | foo2 | 49] Foo2 failed to create class object\n",
+    "12:59:59: [INFO] [foo3.py | foo3 | 43] Foo3 initialized\n",
+    "13:00:12: [ERROR] [foo3.py | foo3 | 30] Foo3 could not be created\n",
+    "13:00:30: [ERROR] [foo3.py | foo3 | 49] Foo3 failed to create class object\n",
+]
+SORTED_LOG_ENTRIES = sorted(UNSORTED_LOG_ENTRIES)
+
+
+@pytest.fixture(name="temp_path")
+def fixture_temp_path() -> pathlib.Path:  # type: ignore
+    """
+    Returns the path to a temporary log directory with dummy log subdirectories.
+    """
+    # Create a temporary logs directory
+    temp_logs_path = TEMP_LOGS_DIRECTORY
+    temp_logs_path.mkdir(parents=True, exist_ok=True)
+
+    # Create sample directories with timestamped names
+    (temp_logs_path / "2024-10-10_10-00-00").mkdir()
+    (temp_logs_path / "2024-10-11_10-00-00").mkdir()
+    (temp_logs_path / "2024-10-09_10-00-00").mkdir()
+
+    yield temp_logs_path
+
+    # Remove the temporary logs directory
+    shutil.rmtree(temp_logs_path)
+
+
+@pytest.fixture(name="dummy_logs")
+def fixture_dummy_logs() -> pathlib.Path:  # type: ignore
+    """
+    Returns the path to a temporary directory with dummy log files.
+    """
+    subdirectory = TEMP_LOGS_DIRECTORY / "subdirectory"
+    subdirectory.mkdir(parents=True, exist_ok=True)
+
+    # Create three dummy log files in the subdirectory
+    log_file_1 = subdirectory / "log1.log"
+    log_file_2 = subdirectory / "log2.log"
+    log_file_3 = subdirectory / "log3.log"
+
+    log_file_1.write_text(
+        "12:59:28: [INFO] [foo1.py | foo1 | 43] Foo1 initialized\n"
+        "13:00:04: [ERROR] [foo1.py | foo1 | 30] Foo1 could not be created\n"
+        "13:00:22: [ERROR] [foo1.py | foo1 | 49] Foo1 failed to create class object\n",
```

Comment:
These can be moved to global constants. Then combined into the unsorted logs with the `+` concatenate operator.